pkg/controller/flink/client/error_handler.go

```diff
@@ -63,3 +69,46 @@ func GetErrorKey(error error) string {
 	}
 	return ""
 }
+
+// A Retryer that has methods to determine if an error is retryable and also does exponential backoff
+type RetryHandler struct {
+	baseBackOffMillis int32
+	timeToWaitMillis  int32
+}
+
+func NewRetryHandler(baseBackoff int32, timeToWait int32) RetryHandler {
+	return RetryHandler{baseBackoff, timeToWait}
+}
+func (r RetryHandler) IsErrorRetryable(err string) bool {
+	if _, ok := retryableErrors[err]; ok {
```

Comment:
This worries me a bit, because typo'ing the name in this map (or updating the error message but not updating the map) would cause this to silently fail. It might be less error-prone to have the map include all errors along with a boolean indicating whether they're retryable; then we could error from this method if an exception isn't found.

Reply:
Enumerating all possible errors (method * httpErrorCodes) is perhaps going to be too long a list? My intention was to enumerate the errors that we know should be retryable or fail fast. On encountering errors that are of neither type, we default to waiting to see if the application makes progress for X minutes. I definitely get your point that we now have to remember to update these two maps. An alternative I can think of is to make the [GetError()](https://github.com/lyft/flinkk8soperator/blob/improve-staleness-logic/pkg/controller/flink/client/error_handler.go#L48) call to also update the map(s). Wdyt?

Reply:
The need for this should no longer exist as the maps have now been removed.